core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -238,29 +237,46 @@ internal class CacheFallbackOperations
         }
 
         public static AdalResultWrapper FindMsalEntryForAdal(ITokenCacheAccessor tokenCacheAccessor, string authority,
-            string clientId, string upn)
+            string clientId, string upn, RequestContext requestContext)
```

Comment:
We should get into the habit of adding comments to all public and internal methods (public for customers, internal for the team).

Reply:
public methods should have docs.
I am totally disagree that we should write comments to all methods.

1. Comments typically used as a crutch for bad code, easily readable code is best "comments".
2. It is quire hard to write good comments, it s harder than write good code. Most of time "comments" are not just useless, very often they are misleading or not synced with code.

Reply:
Sure, I agree that good code is self documenting and I am not saying we should document every parameter etc. The fact that writing good comments is hard is proof that there is value in it. 

While you know all the technical decisions made around using MSAL + ADAL caches, others that will contribute after you (including your future self in 6 months+) do not know this. 

For example for the method above, it is not clear from the method signature what happens if the find operation fails - do you return a null? Does the wrapper encapsulate the possibility of not having a result?

Reply:
this one is pending a follow up discussion on code style

---
In reply to: [199526233](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199526233) [](ancestors = 199526233)

Reply:
pls. take in the code style discussion

---
In reply to: [199973661](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199973661) [](ancestors = 199973661)

Reply:
pls. take in the code style discussion

---
In reply to: [200073976](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r200073976) [](ancestors = 200073976)